chutney/ui/src/app/modules/scenarios/components/execution/history/scenario-executions-history.component.ts

```diff
@@ -293,6 +293,11 @@ export class ScenarioExecutionsHistoryComponent implements OnInit, OnDestroy {
 
     replay(executionId: number) {
         const execution = this.executions.find(exec => exec.executionId === executionId);
-        this.executeScenario(execution.environment, execution.dataset)
+        this.scenarioExecutionService.findExecutionReport(execution.scenarioId, execution.executionId).pipe(
+            map(scenarioExecutionReport => {
+                const dataset = scenarioExecutionReport.datasetId || scenarioExecutionReport.constants || scenarioExecutionReport.datatable ? new Dataset("", "", [], new Date(), scenarioExecutionReport.constants, scenarioExecutionReport.datatable, scenarioExecutionReport.datasetId) : null;
```

Comment:
maybe in backend return null **if not** ` scenarioExecutionReport.datasetId || scenarioExecutionReport.constants || scenarioExecutionReport.datatable`